src/components/Paymethods.tsx

```diff
@@ -0,0 +1,73 @@
+import { number } from "io-ts";
+import { Integer } from "italia-ts-commons/lib/numbers";
+import React, { useState } from "react";
+import { TabContent, TabPane, Nav, NavItem, NavLink } from "reactstrap";
+import { format, parseISO } from "date-fns";
+import { useTranslation } from 'react-i18next';
+import "./Paymethods.css";
+import {PaymentMethod} from "../generated/definitions/PaymentMethod";
+
+var classNames = require('classnames');
+
+function Paymethods(props) {
+
+  const [activeTab, setActiveTab] = useState<number>(0);
+
+  const { t, i18n } = useTranslation();
+
+  const toggle = (tab: number) => {
+    if(activeTab !== tab) setActiveTab(tab);
+  }
+
+    return (
+      <>
+        <div className="mt-4 mb-2 font-weight-bold">
+        {t('Payment methods')}
+        </div>
+
+        <Nav tabs className="MetodiPagamento__list">
+        {props.paylist.map((el: PaymentMethod, index: number) => (
+          <NavItem key={index}>
+          <NavLink
+            className={classNames({
+              'd-flex align-items-center position-relative': true,
+              'active': activeTab==index
+            })}
+            title="Visualizza dettaglio"
+            onClick={()=> {toggle(index)}}
+          >
+            <div className={`MetodiPagamento__stato ${el.payment_instrument_status}`}></div>
+            <div>
+              <div>*{el.payment_instrument_hpan.slice(-5)}</div>
+            </div>
+          </NavLink>
+          </NavItem>
+        ))}
+        </Nav>
+        <TabContent activeTab={activeTab}>
+        {props.paylist.map((el: PaymentMethod, index: number) => (
+          <TabPane tabId={index} key={index}>
```

Comment:
can we use a nested component? (please leave conversations open until solved :)

Reply:
I cannot figure out how transform this react-strap structure in nested components...

Reply:
Can we use the following approach if it works?
```jsx
<TabContent activeTab={activeTab}>
  {props.paylist.map((el: PaymentMethod, index: number) => (
    <PaymentMethodTabPane
      index={index}
      element={el}
    ></PaymentMethodTabPane>
  ))}
</TabContent>
```
With a `PaymentMethodTabPane.tsx`
```jsx
type PaymentMethodTabPaneProps = {
  index: number;
  element: PaymentMethod;
};

function PaymentMethodTabPane(props: PaymentMethodTabPaneProps): JSX.Element {
  const { t, i18n } = useTranslation();
  return (
    <TabPane tabId={props.index} key={props.index}>
      <div className="container my-3">
        <div className="row">
          <div className="col-sm-2 font-weight-bold">{t("State")}</div>
          <div className="col-sm-10">
            {props.element.payment_instrument_status}
....
```

Reply:
yup, that's what I meant :)